pkg/pod/pod.go

```diff
@@ -283,6 +283,55 @@ func (builder *Builder) WaitUntilRunning(timeout time.Duration) error {
 	return builder.WaitUntilInStatus(corev1.PodRunning, timeout)
 }
 
+// WaitUntilHealthy waits for the duration of the defined timeout or until the pod is healthy.
+// An healthy pod is in running phase and optionally in ready condition.
+//
+// timeout is the duration to wait for the pod to be healthy
+// includeSucceeded when true, implies that pod in succeeded phase is running.
+// checkReadiness when true, to also checks that the podCondition is ready.
+// ignoreFailedPods when true, to Ignore failed pod with restart policy set to never.
+func (builder *Builder) WaitUntilHealthy(timeout time.Duration, includeSucceeded, checkReadiness,
+	ignoreFailedPods bool) error {
+	statusesChecked := []corev1.PodPhase{corev1.PodRunning}
+
+	// Ignore failed pod with restart policy never. This could happen in image pruner or installer pods that
+	// will never restart. For those pods, instead of restarting the same pod, a new pod will be created
+	// to complete the task.
+	if ignoreFailedPods &&
+		builder.Object.Status.Phase == corev1.PodFailed &&
+		builder.Object.Spec.RestartPolicy == corev1.RestartPolicyNever {
+		glog.V(100).Infof("Ignore failed pod with restart policy never. Message: %s",
+			statusesChecked, builder.Object.Status.Message)
```

Comment:
The message and objects to format it with don't line up

Reply:
fixed